engines/wintermute/base/scriptables/script_engine.cpp

```diff
@@ -433,7 +433,6 @@ int ScEngine::getNumScripts(int *running, int *waiting, int *persistent) {
 			numPersistent++;
 			break;
 		default:
-			warning("ScEngine::GetNumScripts - unhandled enum");
```

Comment:
Why? I'm not sure that "there's a lot of these messages with verbose debugging on" is a good reason.
Perhaps we need one more warning level?

Reply:
If that message is there for debug reasons only, [debugN()](https://github.com/scummvm/scummvm/blob/d2f5023ee9ec5534a9f7a4754c4f1e805c0fb605/common/debug.h#L33) could be used instead

Reply:
getNumScripts() is actually used only once, inside BaseGame::displayDebugInfo(), when displaying number of running/waiting/persistent threads:
		scrTotal = _scEngine->getNumScripts(&scrRunning, &scrWaiting, &scrPersistent);
		sprintf(str, "Running scripts: %d (r:%d w:%d p:%d)", scrTotal, scrRunning, scrWaiting, scrPersistent);
In ScummVM:
https://github.com/scummvm/scummvm/blob/d4679a4cc4318c0a581f9a823730d6adcfa0b74e/engines/wintermute/base/base_game.cpp#L3777
In WME1:
https://github.com/retrowork/Wintermute-Engine/blob/663d154b93de87f8cd13c86f34345e38a1e727d3/src/engine_core/wme_base/BGame.cpp#L5243

So, entire purpose of this switch block is to calculate some numbers to be shown on debug screen, it is not used for any game logic stuff.

Now, let's check original WME1 implementation: https://github.com/retrowork/Wintermute-Engine/blob/master/src/engine_core/wme_base/ScEngine.cpp#L528

As you can see, there was no "default:" block at all. Unhandled enum elements was the original intended behaviour. 
"default:" block and warning() were added in 4c61cf042b09a64593a8beb6e1e9d72a883a7cda for silincing a compilation warning.

I guess we should either remove this warning, because it does not provide any useful information, we already know that there are some states not listed in sprintf's line, or we can extend this debug information line printing code to include counters for all the states and handle all the enum states.

Reply:
How does the explanation above map to the "tons of warnings" mentioned in the commit message?

If this is purely for debug, then at a minimum a comment, but possibly a debugN-message would be better?

Reply:
> How does the explanation above map to the "tons of warnings" mentioned in the commit message?

Well, method getNumScripts() is called for every frame, producing a warning for every script in those states: SCRIPT_FINISHED, SCRIPT_ERROR, SCRIPT_WAITING_SCRIPT, SCRIPT_THREAD_FINISHED. For subject games there are some scripts with such states, so warnings are constantly printed to Windows console (every frame?), which noticeably slows the game down because of "tons of warnings" (have not checked, but my guess is that SCRIPT_WAITING_SCRIPT is the reason).

> If this is purely for debug, then at a minimum a comment 

I agree - this topic appeared to be way too hot for silently removing warning. =)
I will add a comment once we decide what to do.

>possibly a debugN-message would be better?

I could replace warning() with debugN() or leave warning() as is, but I fail to see a scenario, where debugN() would be useful here. Would we gain any useful information from log lines like "unhandled enum" in this context?

A. If we consider some/all states are dangerous/suspicious/should be traced for some reason - it would be better to add debugN logs on entering/leaving those states (which has nothing to do with getNumScripts() function).

B. If we need a full log of which exact script was in which exact state on which exact game tick - then a more detailed state logger should be implemented (and I'm not sure that getNumScripts() is the right place for this per-tick logger).

C. If we need to see the amount of scripts that are in those states - getNumScripts() should be extended so that more information is added to sprintf() and displayed on screen (that will make ScummVM's debug mode slightly different from original WME code, but why not).

D. If we just need to check that this `switch()` code is imperfect by knowing that there are some states that are unhandled - well, we already know it without reading any logs. We could just leave a comment inside the "default:" block.

Reply:
I would go with D. If the original did not handle such cases, then there is no reason to report them. So, the best scenario would be to add the warning as a debugN() statement, in case it’s needed for debugging.

In any case, a comment should be added, specifying that the default case is unhandled in the original.

Reply:
I second `debugN()` + comment.

Reply:
Done at 49a105b091bc29890ef5bb7557cfcb84a6a19174